.github/workflows/health_checks.yml

```diff
@@ -75,27 +185,40 @@ jobs:
       - uses: ./.github/actions/restore_build_cache
         with:
           node-version: ${{ matrix.node }}
+          cdk-version: ${{ needs.resolve_inputs.outputs.cdk_version }}
       - run: npm run set-script-shell
       - run: npm run test:coverage:threshold
   test_scripts:
     needs:
       - build
+      - resolve_inputs
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # version 4.1.4
       - uses: ./.github/actions/setup_node
+        with:
+          node-version: ${{ fromJSON(needs.resolve_inputs.outputs.node)[0] }}
```

Comment:
Should we be explicit here instead of just getting the first version?

Reply:
To do this I would need to convert `NodeJS array` input to `Include Node 20?` and have 18 always included by default.

Otherwise this would fail if workflow was ran with `["20"]` exclusively because install and build cache would be not present.

That's another aspect I was a bit unsure what to do so I left it flexible.

wdyt?

Reply:
Got it, can we run the build/install for 18,20 regardless every single time?. I'm worries that [18,20] and [20,18] will behave differently and will be harder to debug.

Reply:
we can do that too.

Reply:
Done.

see https://github.com/aws-amplify/amplify-backend/actions/runs/12361549810